pkg/imgpkg/bundle/bundle.go

```diff
@@ -132,6 +133,21 @@ func (o *Bundle) pull(baseOutputPath string, ui goui.UI, pullNestedBundles bool,
 		ui.BeginLinef("Pulling nested bundle '%s'\n", o.DigestRef())
 	}
 
+	bundleDigestRef, err := regname.NewDigest(o.plainImg.DigestRef())
+	if err != nil {
+		return err
+	}
+
+	loggerBuilder := util.NewLogger(uiBlockWriter{ui})
+	locationFetcher := cachedLocationFetcher{
+		LocationFetcher{
+			logger:          loggerBuilder.NewLevelLogger(util.LogWarn, loggerBuilder.NewPrefixedWriter("")),
```

Comment:
Yeah good call out. I agree. I'll leave it to a different PR since it effects copy.

Also, I wonder if we shouldn't use a logger, and instead use `ui`. This would involve changing copy from `logger` to `ui`.